test/components/plots/barPlotTests.ts

```diff
@@ -355,6 +391,42 @@ describe("Plots", () => {
         assert.closeTo(numAttr(bar1, "y"), yScale.scale(bar1y) - numAttr(bar1, "height") / 2, 0.01, "bar1 ypos");
         svg.remove();
       });
+
+      it("getAllPlotData() pixel points corrected for negative-valued bars", () => {
+        var plotData = barPlot.getAllPlotData();
+        plotData.data.forEach((datum, i) => {
+          var barSelection = d3.select(plotData.selection[0][i]);
+          var pixelPointX = plotData.pixelPoints[i].x;
+          if (datum.x < 0) {
+            assert.strictEqual(pixelPointX, +barSelection.attr("x"), "negative on left");
+          } else {
+            assert.strictEqual(pixelPointX, +barSelection.attr("x") + +barSelection.attr("width"), "positive on right");
+          }
+        });
+        svg.remove();
+      });
+
+      it("getAllPlotData() pixel points corrected for barAlignment left", () => {
+        barPlot.barAlignment("left");
```

Comment:
We used to have "top" "center" "bottom" semantics but to keep semantics consistent, we switched to the same "left" "right" "center"